code/modules/lighting/lighting_source.dm

```diff
@@ -205,50 +216,98 @@
-/datum/light_source/proc/generate_sheet_multiz(range)
+/datum/light_source/proc/generate_sheet_multiz(range, visual_offset, x_offset, y_offset, center_dir, angle)
 	var/list/encode = list()
 	var/z_range = SSmapping.max_plane_offset // Let's just be safe yeah?
 	for(var/z in -z_range to z_range)
-		var/list/sheet = generate_sheet(range, z)
+		var/list/sheet = generate_sheet(range, visual_offset, x_offset, y_offset, center_dir, angle, z)
 		encode += list(sheet)
 	return encode
 
 /// Takes x y and z offsets from the source as input, alongside our source's range
 /// Returns a value between 0 and 1, 0 being dark on that tile, 1 being fully lit
-/datum/light_source/proc/falloff_at_coord(x, y, z, range)
-	var/_range_divisor = max(1, range)
+/datum/light_source/proc/falloff_at_coord(x, y, z, range, center_dir, angle)
+	var/range_divisor = max(1, range)
+
 	// You may notice we use squares here even though there are three components
 	// Because z diffs are so functionally small, cubes and cube roots are too aggressive
-	return 1 - CLAMP01(sqrt(x ** 2 + y ** 2 + z ** 2 + LIGHTING_HEIGHT) / _range_divisor)
-
+	// The larger the distance is, the less bright our light will be
+	var/multiplier = 1 - CLAMP01(sqrt(x ** 2 + y ** 2 + z ** 2 + LIGHTING_HEIGHT) / range_divisor)
+	if(angle >= 360 || angle <= 0)
+		return multiplier
+
+	// Turn our positional offset into an angle
+	var/coord_angle = delta_to_angle(x, y)
+	// Get the difference between the angle we want, and the angle we have
+	var/center_angle = dir2angle(center_dir)
+	var/angle_delta = abs(center_angle - coord_angle)
+	// Now we have to normalize the angle delta to be between 0 and 180, instead of 0 and 360
+	// This ensures removing say, 15 degrees removes it from both sides, rather then just one
+	// Turns an unfurling fan into a pair of scissors
+	if(angle_delta > 180)
+		angle_delta = 180 - (angle_delta - 180)
+	// We allow angle deltas to a certian amount, angle / 2
+	// If we pass that, then it starts effecting the visuals
+	// Oh and we'll scale it so 30 degrees is the "0" point, where things become fully dark
+	// This could be variable, it just isn't yet yaknow?
+	return max(multiplier * (1 - max(angle_delta - (angle / 2), 0) / 30), 0)
+
+/// Dumps the content of a lighting sheet to chat, for debugging
+/datum/light_source/proc/print_sheet()
+	var/list/sheet = get_sheet()
+	var/list/output = list()
+	var/multiz_depth = 1
+	// If we have a list 3 layers down we're multiz
+	if(length(sheet[1][1]))
+		multiz_depth = length(sheet)
+	var/column_seperator = ""
+	for(var/i in 1 to length(sheet))
+		column_seperator += "----"
+	output += column_seperator
+	for(var/i in 1 to multiz_depth)
+		for(var/list/column in sheet)
+			var/list/print_column = list()
+			for(var/row in column)
+				print_column += round(row, 0.1)
+			output += print_column.Join(", ")
+		output += column_seperator
+	message_admins("\n[output.Join("\n")]")
```

Comment:
message admins seems like it could get annoying in case someone wants to quickly check something wacky on prod during a test merge of a new map or something, would you mind passing in a user or something?

Reply:
I mean sure